prov/rxd/src/rxd_cq.c

```diff
@@ -214,7 +214,7 @@ static void rxd_complete_rx(struct rxd_ep *ep, struct rxd_x_entry *rx_entry)
 	if ((!(rx_entry->cq_entry.flags & FI_REMOTE_CQ_DATA) &&
 	     (rx_entry->cq_entry.flags & FI_RMA ||
 	     rx_entry->cq_entry.flags & FI_ATOMIC)) ||
-	     rx_entry->flags & RXD_NO_RX_COMP)
+	     rx_entry->flags & RXD_NO_RX_COMP || rx_entry->flags & RXD_CANCELLED)
```

Comment:
This can be simplified as
```
rx_entry->flags & (RXD_NO_RX_COMP | RXD_CANCELED)
```
A similar change can be made to cq_entry.flags check above it.

Reply:
Btw, how does cancel differ from the no_rx_comp flag?

Reply:
The cancel flag is checked in progress_op to increment the sequence numbers for the entire canceled message so we can account for them without having to process them while the no_rx_comp_flag still processes the message just without generating a completion.